proposals/arbitrary-broker-resources.md

```diff
@@ -0,0 +1,132 @@
+# Deploy arbitrary Kubernetes resources as broker
+
+## Background
+
+Currently, a broker is defined as either a Job or a Pod, and can be accompanied by a Configuration level Service and an Instance level service, as shown in following YAML example.
+
+```yaml
+apiVersion: akri.sh/v0
+kind: Configuration
+metadata:
+  name: akri-debug-echo-foo
+spec:
+  brokerSpec:
+    brokerPodSpec:
+      containers:
+      - name: akri-debug-echo-foo-broker
+        image: "nginx:stable-alpine"
+        resources:
+          requests:
+            "{{PLACEHOLDER}}" : "1"
+            memory: 10Mi
+            cpu: 10m
+          limits:
+            "{{PLACEHOLDER}}" : "1"
+            memory: 30Mi
+            cpu: 29m
+  instanceServiceSpec:
+    type: ClusterIP
+    ports:
+    - name: akri-debug-echo-foo-instance-service
+      port: 6052
+      protocol: TCP
+      targetPort: 6052
+  configurationServiceSpec:
+    type: ClusterIP
+    ports:
+    - name: akri-debug-echo-foo-configuration-service
+      port: 6052
+      protocol: TCP
+      targetPort: 6052
+```
+
+This limits the interoperability of Akri with other Kubernetes based components that would benefit from Akri creating CRs or other Kubernetes native resources.
+For example Akri could act as the discovery component of a larger system that leverage Digital Twins with [Shifu](https://shifu.dev/).
+
+## New broker definition
+
+The idea behind the new broker definition is to provide two fields `instanceBrokerResources` and `configurationBrokerResources` that will contain a list of Kubernetes resources to deploy.
+
+Instance level resources will get deployed for every instance, and Configuration level ones will get deployed once for a Configuration with at least a discovered Instance.
+
+Akri will still add Configuration and Instance labels to resources it creates.
+
+## New template variables
+
+All those possibilities requires something a bit more advanced than the current `{{PLACEHOLDER}}` replacement, as the resources might need additional fields like the Instance name, or some of its properties.
+To cover this let's have a couple of other "variables" that would replace the `PLACEHOLDER` one:
+
+| Variable | Description |
+| -------- | ----------- |
+| RESOURCE_ID | The string to put in resource requests (i.e. the current PLACEHOLDER) |
+| INSTANCE_NAME | The name of the instance |
```

Comment:
Just to confirm, the difference between RESOURCE_ID and INSTANCE_NAME is the RESOURCE_ID has additional prefix "akri.sh/".  If this is true, can we consolidate them to one variable?